### PR TITLE
Make community build docs experiment-aware

### PR DESCRIPTION
**Pull request by abgruszecki**

### CODE REVIEW

**Comment by abgruszecki** [comment]
Actually tested in  #11936.

**Comment by abgruszecki** [comment]
Ok, this seems to work as intended, the parts of the tests that fail are the ones that fail on master as well.

**Comment by abgruszecki** [comment]
@BarkingBad Krzysztof mentioned that you'll be looking at the CI problems with the doctool, correct? Could you ping me when you're done so that I can rebase this PR?

**Comment by BarkingBad** [comment]
Hey @abgruszecki, I think it was becuase of this nasty bug #11948.

Becuase of that, the `useScaladoc.value` in `dotty/sbt-dotty/sbt-test/sbt-dotty/scaladoc/build.sbt` was returning false (becuase version is `3.0.1`), hence the aseertion was failing without meaningful comment.

**Comment by BarkingBad** [comment]
Oh I didn't see you already fixed that. I think you can rebase now to make the CI pass. You can close my PR or I can fix it to add that assertion message for future debugging.

**Comment by abgruszecki** [comment]
@odersky this PR adds support for `compilerSupportExperimental` in the doc part of the community build. You can see in #11936 that the doctool community build passes if we simulate an experimental-unsupporting build. There is a single compiler test that fails when we do that, but solving that is a separate issue:
```
9 suites passed, 1 failed, 10 total
    tests/plugins/neg/divideZero-research failed
```

**Review by odersky — COMMENTED** [review]
I am a bit surprised that so many projects require experimental. How did they build before? 

I would be good if @nicolasstucki or @liufengyun could review as well.

File: community-build/src/scala/dotty/communitybuild/projects.scala [hunk 1]
```diff
@@ -237,13 +246,15 @@ object projects:
 
   lazy val scas = MillCommunityProject(
     project = "scas",
-    baseCommand = "scas.application"
+    baseCommand = "scas.application",
+    requiresExperimental = true
```
Comment (liufengyun) [inline community-build/src/scala/dotty/communitybuild/projects.scala:250]: `scas` does not seem to depend on experimental.

**Review by liufengyun — APPROVED** [review]

File: community-build/src/scala/dotty/communitybuild/projects.scala (hunk 1 above)
Comment (abgruszecki) [reply community-build/src/scala/dotty/communitybuild/projects.scala]: Yep, looking at the commit history that's correct; this was an oversight. Other projects required experimental previously as far as I can tell.

**Pull request merged**